app/scripts/metamask-controller.js

```diff
@@ -2115,15 +2121,19 @@ export default class MetamaskController extends EventEmitter {
   }
 
   triggerNetworkrequests() {
-    this.accountTracker.start();
-    this.txController.startIncomingTransactionPolling();
-    if (this.preferencesController.store.getState().useCurrencyRateCheck) {
-      this.currencyRateController.startPollingByNetworkClientId(
-        this.networkController.state.selectedNetworkClientId,
-      );
-    }
-    if (this.preferencesController.store.getState().useTokenDetection) {
-      this.tokenListController.start();
+    if (!this.hasNetworkRequestsBeenTriggered) {
```

Comment:
Why was this new if condition needed?

Reply:
If the user unlocks the wallet, locks the wallet, and then unlocks the wallet again, we don't want an unnecessary doubling of the requests being triggered. This condition ensures that we keep track of wether or not the polling has already been started.